frontend/src/components/homepage/homepage.tsx

```diff
@@ -42,9 +49,31 @@ const Homepage: React.FC = () => {
         search,
       });
     }, SEARCH_TIMEOUT),
-    [],
+    [podcastsFilter],
   );
 
+  const handleSetGenres = ({ genresFilter }: GenresFilter): void => {
+    const selectedGenres = genres.reduce<number[]>((selectedGenres, genre) => {
+      if (genresFilter[genre.key]) {
+        selectedGenres.push(genre.id);
+      }
+
+      return selectedGenres;
+    }, []);
```

Comment:
```suggestion
    const selectedGenres = genres.reduce<number[]>((acc, genre) => {
    const hasGender = Boolean(genresFilter[genre.key]);
      return hasGender ? acc.concat(genre.id) ? acc;
    }, []);
```
create a helper from it and move to frontend/src/components/homepage/helpers